<launch>
  <!-- Run rosserial for the arduino file uploaded --> 
  <!-- Wheel Encoder Tick Publisher and Base Controller Using Arduino --> 
  <!-- motor_controller_diff_drive_2.ino is the Arduino sketch -->
  <!-- Subscribe: /cmd_vel -->
  <!-- Publish: /right_ticks, /left_ticks -->
  <node pkg="rosserial_python" type="serial_node.py" name="serial_node">
    <param name="port" value="/dev/ttyACM0"/>
    <param name="baud" value="57600"/>
  </node>
 
  <!-- Run the odometry.py file -->
  <!-- Wheel Odometry Publisher -->
  <!-- Subscribe: /lwheel, /rwheel -->
  <!-- Publish: /odom -->
  <rosparam param="ticks_meter">10000</rosparam>
  <node pkg="digitallab" type="diff_drive_odometry.py" name="diff_drive_odometry" output="screen">
    <rosparam param="ticks_meter">10000</rosparam>
    <rosparam param="base_width">0.329</rosparam>
  </node>

</launch>
